File: backend/lms/src/main/java/com/el/coursepath/application/dto/CoursePathPublishedDto.java
package com.el.coursepath.application.dto;

import javax.money.MonetaryAmount;
import java.time.LocalDateTime;
import java.util.Set;

public record CoursePathPublishedDto(
        Long id,
        String title,
        String description,
        Set<CourseOrderPublishedDto> courseOrders,
        String teacher,
        LocalDateTime publishedDate
) {
    public record CourseOrderPublishedDto(
            Long id,
            Long courseId,
            Integer orderIndex,
            MonetaryAmount price,
            String title,
            Integer purchaseCount
    ) {
        public CourseOrderPublishedDto addPurchaseCount(int purchaseCount) {
            return new CourseOrderPublishedDto(
                    id,
                    courseId,
                    orderIndex,
                    price,
                    title,
                    purchaseCount
            );
        }
    }
}

File: angular-ui/src/app/administration/course-path/model/course-path-published.dto.ts
export interface CourseOrderPublishedDto {
  id: number;
  courseId: number;
  orderIndex: number;
  price: string;
  title: string;
  purchaseCount: number;
}

export interface CoursePathPublishedDto {
  id: number;
  title: string;
  description: string;
  courseOrders: CourseOrderPublishedDto[];
  teacher: string;
  publishedDate: string;
}
